django_freeradius/api/views.py

```diff
@@ -137,3 +139,81 @@ def _get_status_type(self, request):
 
 
 accounting = AccountingView.as_view()
+
+
+class BatchPrefixView(generics.CreateAPIView):
+
+    queryset = RadiusBatch.objects.all()
+    serializer_class = RadiusBatchPrefixSerializer
+
+    def post(self, request, *args, **kwargs):
+        serializer = RadiusBatchPrefixSerializer(data=request.data)
+        if serializer.is_valid():
+            name = serializer.data.get('name')
+            strategy = self._get_strategy(request)
+            expiration_date = serializer.data.get('expiration_data')
+            prefix = serializer.data.get('prefix')
+            batch = RadiusBatch(name=name, strategy=strategy, expiration_date=expiration_date, prefix=prefix)
+            number_of_users = self._get_number_of_users(request)
+            batch.prefix_add(prefix, number_of_users)
+            response = RadiusBatchPrefixSerializer(batch)
+            return Response(response.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def _get_strategy(self, request):
+        try:
+            s = request.data['strategy']
```

Comment:
please use readable variable names, change `s` to `strategy`, keep in mind code is read a lot more than is written (and the one reading this code over and over will be more likely me :-P)